File: client/src/app/_models/recipe.ts
import { DifficultyLevelEnum } from './difficultyLevelEnum';
import { Instruction } from './instruction';
import { Photo } from './photo';
import { PreparationTimeEnum } from './preparationTimeEnum';
import { Rate } from './rate';
import { RecipeIngredient } from './recipeIngredient';

export interface Recipe {
  id: number | null;
  name: string;
  description?: string;
  userId: number;
  author: string;
 
  categoryId: number;
  category: string;
  difficultyLevel: DifficultyLevelEnum;
  //difficulty: string;
  preparationTime: PreparationTimeEnum;
  //preparation: string;
  instructions: Instruction[];
  photos: Photo[];
  recipeIngredients: RecipeIngredient[];
  countCookbooks: number;

  rateValues: number;
  countRates: number;
  rates: Rate[];
  totalScore: number;
}

File: CulinaryPortal.Domain/Entities/Recipe.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using CulinaryPortal.Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace CulinaryPortal.Domain.Entities
{
    public class Recipe
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(30)]
        public string Name { get; set; }        

        [MaxLength(100)]
        public string Description { get; set; }

        public int? UserId { get; set; }
        public User User { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; }
        [Required]
        [JsonConverter(typeof(StringEnumConverter))]
        public DifficultyLevel DifficultyLevel { get; set; }

        [Required]
        [JsonConverter(typeof(StringEnumConverter))]
        public PreparationTime PreparationTime { get; set; }

        public IList<Instruction> Instructions { get; set; }
        
        public IList<Photo> Photos { get; set; }

        public IList<CookbookRecipe> CookbookRecipes { get; set; }// = new List<CookbookRecipe>();

        public IList<RecipeIngredient> RecipeIngredients { get; set; }

        public IList<Rate> Rates { get; set; }
    }
}
